File: CWSpectrumDllTest/frmTestOxyPlot.vb
Imports OxyDataPlotter

Public Class frmTestOxyPlot

    Private WithEvents mOxyPlot As frmOxySpectrum

    Private Sub InitializeOxySpectrum()

        Try
            mOxyPlot = New frmOxySpectrum

            mOxyPlot.Text = "Test Plot"

            mOxyPlot.Show()

        Catch ex As Exception
            MessageBox.Show("Error initializing CWSpectrumDLLNET.frmOxySpectrum: " & ex.Message)
        End Try
    End Sub

    Private Sub frmTestCWSpectrum_Closing(sender As Object, e As System.ComponentModel.CancelEventArgs) Handles MyBase.Closing
        mOxyPlot = Nothing
    End Sub

    Private Sub cmdExit_Click(sender As System.Object, e As System.EventArgs) Handles cmdExit.Click
        Try
            Me.Close()
        Catch ex As Exception
            MessageBox.Show("Exception in cmdExit_Click: " & ex.Message)
        End Try
    End Sub

    Public Sub New()

        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        InitializeOxySpectrum()
    End Sub

    Private Sub cmdShowPlot_Click(sender As System.Object, e As EventArgs) Handles cmdShowPlot.Click
        If mOxyPlot Is Nothing Then
            InitializeOxySpectrum()
        Else
            mOxyPlot.Show()
        End If

    End Sub
End Class

File: OxyPlotterTest/frmTestOxyPlot.cs
using System;
using System.Windows.Forms;

namespace OxyPlotterTest
{
    public partial class frmTestOxyPlot : Form
    {
        private OxyDataPlotter.frmOxySpectrum mOxyPlot;

        private Timer mSetFocusTimer;

        public frmTestOxyPlot()
        {
            InitializeComponent();

            InitializeOxySpectrum();

            mSetFocusTimer = new Timer {
                Interval = 250
            };

            mSetFocusTimer.Tick += SetFocusTimer_Tick;
            mSetFocusTimer.Enabled = true;
        }

        private void InitializeOxySpectrum()
        {
            try
            {
                mOxyPlot = new OxyDataPlotter.frmOxySpectrum {
                    Text = "Test Plot"
                };

                mOxyPlot.Show();
            }
            catch (Exception ex)
            {
                System.Windows.Forms.MessageBox.Show("Error initializing CWSpectrumDLLNET.frmOxySpectrum: " + ex.Message);
            }
        }

        private void cmdShowPlot_Click(object sender, EventArgs e)
        {
            if (mOxyPlot == null)
                InitializeOxySpectrum();
            else
                mOxyPlot.Show();
        }

        private void cmdExit_Click(object sender, EventArgs e)
        {
            try
            {
                this.Close();
            }
            catch (Exception ex)
            {
                System.Windows.Forms.MessageBox.Show("Exception in cmdExit_Click: " + ex.Message);
            }
        }

        private void SetFocusTimer_Tick(object sender, EventArgs e)
        {
            mOxyPlot.BringToFront();
            mSetFocusTimer.Enabled = false;
        }

    }
}
